### ISSUE TITLE
Implement hathor-core messages integration with AWS

### ISSUE THREAD

**Issue by pedroferreira1** [HathorNetwork/hathor-wallet-service-old#2]
PRs:

- https://github.com/HathorNetwork/hathor-core/pull/99
- https://github.com/HathorNetwork/hathor-wallet-service/pull/3

# Problem

The wallet service needs to update its data when a transaction is created or updated on the network. The full node already has websocket messages for those events, so the wallet service could just use them.

# Possible solutions

1. Use an AWS service to listen to those ws messages

There is no way to do this directly using a service, only if we use a third party app also (e.g. https://fanout.io/) that listens for ws messages and triggers a HTTP request.

2. Full node add the ws message also on a SQS and the lambda reads from it

A flag would be added to the full node (e.g. `--sqs-key`) as a key to add the messages to the SQS. A class would be created to listen from the metrics object and add the messages to the SQS (like we do in the Prometheus class).

The problems I see with this solutions are (i) we add another requirement to the full node (aws lib) that has nothing to do with it and (ii) we should run this in only one full node, otherwise we would be adding repeated messages to the queue, what would increase the complexity of the deploy of new versions.

3. A separated service in nodejs that is a ws client that listens to the messages and triggers the lambdas

The problem with this solution is to have another service to maintain, debug and deploy.

--- 

In my opinion option 1 is the worst and we should keep the full node as clean as possible. So I think we should go with solution 3.

If we decide to go with solution 2 or 3, it's important to try to fix the duplicated ws messages bug on the full node (https://github.com/HathorNetwork/hathor-core/issues/73).

**Comment by msbrogli** [comment]
I agree we should go with solution 3; and I think it should run in the same host as hathor-wallet-service.

**Comment by jansegre** [comment]
I like 3 the best too.

**Comment by obiyankenobi** [comment]
Yeah, also agree to go with 3.

> and I think it should run in the same host as hathor-wallet-service.

@msbrogli I'm not sure what this means as we're using serverless for the wallet-service.
